src/main/java/org/datadog/jenkins/plugins/datadog/DatadogBuildListener.java

```diff
@@ -221,7 +248,40 @@ public final void onCompleted(final Run run, @Nonnull final TaskListener listene
       logger.fine("Finished onCompleted()");
     }
   }
-  
+
+  private long getMeanTimeBetweenFailure(final Run run) {
+    Run<?, ?> lastGreenRun = run.getPreviousNotFailedBuild();
+    if (lastGreenRun != null) {
+      return run.getStartTimeInMillis() - lastGreenRun.getStartTimeInMillis();
+    }
+    return 0;
+  }
+
+  private long getCycleTime(final Run run) {
+    Run<?, ?> previousSuccessfulBuild = run.getPreviousSuccessfulBuild();
+    if (previousSuccessfulBuild != null) {
+      return (run.getStartTimeInMillis() + run.getDuration()) -
+              (previousSuccessfulBuild.getStartTimeInMillis() + previousSuccessfulBuild.getDuration());
+    }
+    return 0;
+  }
+
+  private long getMeanTimeToRecovery(final Run run) {
+    if (buildFailed(run.getPreviousBuiltBuild())) {
+      Run<?, ?> firstFailedRun = run.getPreviousBuiltBuild();
+
+      while (buildFailed(firstFailedRun.getPreviousBuiltBuild())) {
```

Comment:
Could this lead to an infinite loop? Should we have a max just in case?

Reply:
I took a look at the Java docs (as I had a similar thought) and my read was that we'd stop at the root if there were no failed runs (which seems a bit unlikely to me though)

Reply:
What about performance, could it be that it loops for so long that it makes the all thing slow. Maybe too far fetched. You tell me.